src/topology/vector_bundle/basic.lean

```diff
@@ -45,312 +45,230 @@ Vector bundle
 
-end topological_vector_bundle.pretrivialization
+variables (R) [topological_space (total_space E)]
 
-variable [topological_space (total_space E)]
+/-- A mixin class for `trivialization`, stating that a trivialization is fibrewise linear with
+respect to given module structures on its fibres and the model fibre. -/
+protected class trivialization.is_linear [add_comm_monoid F] [module R F]
+  [∀ x, add_comm_monoid (E x)] [∀ x, module R (E x)] (e : trivialization F (π E)) : Prop :=
+(linear : ∀ b ∈ e.base_set, is_linear_map R (λ x : E b, (e (total_space_mk b x)).2))
 
-/--
-A structure extending local homeomorphisms, defining a local trivialization of the projection
-`proj : total_space E → B` with fiber `F`, as a local homeomorphism between `total_space E`
-and `B × F` defined between two sets of the form `proj ⁻¹' base_set` and `base_set × F`,
-acting trivially on the first coordinate and linear in the fibers.
--/
-@[ext, nolint has_nonempty_instance]
-structure topological_vector_bundle.trivialization extends to_fiber_bundle_trivialization :
-  topological_fiber_bundle.trivialization F (@total_space.proj B E) :=
-(linear' : ∀ x ∈ base_set, is_linear_map R (λ y : E x, (to_fun (total_space_mk x y)).2))
-
-open topological_vector_bundle
-
-instance : has_coe_to_fun (trivialization R F E) (λ _, total_space E → B × F) := ⟨λ e, e.to_fun⟩
-
-instance : has_coe (trivialization R F E)
-  (topological_fiber_bundle.trivialization F (@total_space.proj B E)) :=
-⟨topological_vector_bundle.trivialization.to_fiber_bundle_trivialization⟩
-
-namespace topological_vector_bundle.trivialization
-
-variables {R F E} (e : trivialization R F E) {x : total_space E} {b : B} {y : E b}
+namespace trivialization
 
-/-- Natural identification as `topological_vector_bundle.pretrivialization`. -/
-def to_pretrivialization (e : trivialization R F E) :
-  topological_vector_bundle.pretrivialization R F E := { ..e }
+variables (e : trivialization F (π E)) {x : total_space E} {b : B} {y : E b}
 
-protected lemma linear (hb : b ∈ e.base_set) :
+protected lemma linear [add_comm_monoid F] [module R F] [∀ x, add_comm_monoid (E x)]
+  [∀ x, module R (E x)] [e.is_linear R] {b : B} (hb : b ∈ e.base_set) :
   is_linear_map R (λ y : E b, (e (total_space_mk b y)).2) :=
-e.linear' b hb
+trivialization.is_linear.linear b hb
 
-protected lemma continuous_on : continuous_on e e.source := e.continuous_to_fun
+instance to_pretrivialization.is_linear [add_comm_monoid F] [module R F]
+  [∀ x, add_comm_monoid (E x)] [∀ x, module R (E x)] [e.is_linear R] :
+  e.to_pretrivialization.is_linear R :=
+{ ..(‹_› : e.is_linear R) }
 
-lemma to_pretrivialization_injective :
-  function.injective (λ e : trivialization R F E, e.to_pretrivialization) :=
-by { intros e e', rw [pretrivialization.ext_iff, trivialization.ext_iff,
-  ← topological_fiber_bundle.trivialization.to_pretrivialization_injective.eq_iff], exact id }
+protected lemma continuous_on : continuous_on e e.source := e.continuous_to_fun
```

Comment:
again, this lemma and several lemmas below (until the section `has_zero`) should be moved to a more proper place as they have nothing to do with linearity. Aren't some of them duplicates of preexisting lemmas, by the way?

Reply:
Same question as [here](https://github.com/leanprover-community/mathlib/pull/17359#discussion_r1015835523) about combined vs separate PRs.